### PR TITLE
Exclude the UID of redirected file from XRef map 

### PR DESCRIPTION
**Pull request by freewheel70**
[AB#539493](https://ceapex.visualstudio.com/Engineering/_workitems/edit/539493)

Exclude the UID of redirected file from XRef map

### CODE REVIEW

File: docs/specs/xref.yml [hunk 1]
```diff
@@ -2162,3 +2162,69 @@ outputs:
   .errors.log: |
     {"message_severity":"warning","code":"missing-attribute","message":"Missing required attribute: 'name'.","file":"docs/TOC.md"}
     {"message_severity":"warning","code":"xref-not-found","message":"Cross reference not found: 'b'.","file":"docs/TOC.md"}
+---
+# files to be redirected must be excluded from xref map
```
Comment (yufeih) [inline docs/specs/xref.yml:2166]:
```suggestion
# exclude redirected files from xref map
```

File: docs/specs/xref.yml [hunk 2]
```diff
@@ -2162,3 +2162,69 @@ outputs:
...
+noSingleFile: true
+repos:
+  https://docs.com/redirected-uid-excluded-from-xrefa#master:
+    - files:
+        docfx.yml:
+        _themes/ContentTemplate/schemas/TestData.schema.json: |
```
Comment (yufeih) [inline docs/specs/xref.yml:2172]: Simplify this test case by assigning uids to markdowns:
```suggestion
       a.md: |
       ---
       uid: a
       ---
```

File: docs/specs/xref.yml [hunk 3]
```diff
@@ -2162,3 +2162,69 @@ outputs:
...
+      "references":[
+        {
+          "uid": "b",
+          "title": "Title b",
+          "name": "name b",
+          "fullName": "fullName b"
+        }
+      ]
+    }
+  b.json: |
```
Comment (yufeih) [inline docs/specs/xref.yml:2210]: Ignore the content of this file. The purpose of the test is to check xrefmap, can simply assert `a` does not exit like so:
```suggestion
  .xrefmap.json: |
     { "references" : [{ "uid": "a" }] }
  b.json:
```

File: docs/specs/xref.yml [hunk 4]
```diff
@@ -2162,3 +2162,69 @@ outputs:
...
+            }
+         },
+       "uid":"b",
+       "fullName":"fullName b",
+       "name":"name b",
+       "schema":"TestData",
+       "title":"Title b",
+     }
+  .errors.log: | 
+    {"message_severity":"warning","code":"redirected-file-not-removed","message":"Redirected file 'a.yml' are still in the repo. After adding a file to the redirection JSON file, you must delete the original file from the repo.","line":0,"end_line":0,"column":0,"end_column":0,"log_item_type":"user"}
```
Comment (yufeih) [inline docs/specs/xref.yml:2230]: Remove properties with default values.
```suggestion
    {"message_severity":"warning","code":"redirected-file-not-removed","message":"Redirected file 'a.yml' are still in the repo. After adding a file to the redirection JSON file, you must delete the original file from the repo."}
```

File: src/docfx/build/xref/InternalXrefMapBuilder.cs [hunk 5]
```diff
@@ -27,7 +28,8 @@ internal class InternalXrefMapBuilder
         BuildScope buildScope,
         RepositoryProvider repositoryProvider,
         Input input,
-        Func<JsonSchemaTransformer> jsonSchemaTransformer)
+        Func<JsonSchemaTransformer> jsonSchemaTransformer,
+        RedirectionProvider redirectionProvider)
```
Comment (yufeih) [inline src/docfx/build/xref/InternalXrefMapBuilder.cs:32]: Put `Func<...>` args at last

**Comment by freewheel70** [comment]
update code according to the comments of current PR and #7819

**Review by live1206 — APPROVED** [review]
👍

**Review by yufeih — APPROVED** [review]

**Review by herohua — APPROVED** [review]

**Pull request merged**